### PR TITLE
Fbttf 155 ia

### PR DESCRIPTION
**Pull request by guillaumeAbel**

### CODE REVIEW

File: src/ennemy/update_ennemy_move.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+*/
+
+#include <stdlib.h>
+#include "ennemies.h"
+#include "ennemy_pathfind.h"
+
+static int get_data(any_t **rect_speed, any_t **move_speed, any_t *data,
+    any_t **rect_list)
+{
+    if (data == NULL) {
```
Comment (Saverio976) [inline src/ennemy/update_ennemy_move.c:15]: check if rect_speed is NULL or move_speed too
because *NULL -> seg

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: src/ennemy/create_ennemy.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        return RET_ERR_INPUT;
+    }
+    ennemy->state = STOP;
+    return set_rect(ennemy, stop_rect->value.array);
+}
+
+static ennemy_t *add_new_ennemy_struct(scene_t *scene, object_t *obj,
+    any_t *ennemy_data)
+{
+    ennemy_t *ennemy = malloc(sizeof(ennemy_t));
```
[the remark this answers is not in the document]
Comment (guillaumeAbel) [reply src/ennemy/create_ennemy.c:52]: it is a static function so 100% all input are valid and the if after this malloc is only to check this malloc and other allocation made in the same time

File: src/ennemy/update_ennemy_move.c (hunk 1 above)
Comment (guillaumeAbel) [reply src/ennemy/update_ennemy_move.c:15]: there are NULL fill them here

File: src/ennemy/update_ennemy_move.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        *rect_id += 1;
+        if (*rect_id >= rect_list->value.array->len) {
+            *rect_id = 0;
+        }
+    }
+}
+
+void update_ennemy_move(ennemy_t *ennemy, scene_t *scene, window_t *win,
+    float time)
+{
```
[the remark this answers is not in the document]
Comment (guillaumeAbel) [reply src/ennemy/update_ennemy_move.c:77]: update ennemy move is call from update ennemy where ennemy is checked

[the remark this answers is not in the document]
Comment (Saverio976) [reply src/ennemy/update_ennemy_move.c:81]: NAH

**Review by Saverio976 — CHANGES_REQUESTED** [review]

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**